ios/brave-ios/Sources/Brave/Utility/AppDebugComposer.swift

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Data
+import Foundation
+import Preferences
+import UIKit
+import os.log
+
+struct AppDebugComposer {
+
+  private static var printDeviceInfo: String {
+    let device = UIDevice.current
+    let model = device.userInterfaceIdiom == .pad ? "iPad" : "iPhone"
+    let iOSVersion = "\(device.systemName) \(device.systemVersion)"
+    let appVersion =
+      Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? "-"
+
+    return "Setup: \(model), \(iOSVersion), Brave \(appVersion)"
+  }
+
+  /// This function prepares data to help us identify any app storage problems users may have.
+  static func composeAppSize() -> String {
```

Comment:
This method is unchanged, renamed so this class can handle more copy-clipboard actions